src/Microsoft.OData.Client/Serialization/Serializer.cs

```diff
@@ -405,6 +606,132 @@ internal void WriteNestedResourceInfo(EntityDescriptor entityDescriptor, IEnumer
             }
         }
 
+        /// <summary>
+        /// Writes a nested resource
+        /// </summary>
+        /// <param name="entityDescriptor">The entity</param>
+        /// <param name="relatedLinks">The links related to the entity</param>
+        /// <param name="odataWriter">The ODataWriter used to write the navigation link.</param>
+        /// <param name="bulkUpdateGraph">An instance of the <see cref="BulkUpdateGraph"/></param>
+        internal void WriteDeltaNestedResourceInfo(EntityDescriptor entityDescriptor, BulkUpdateGraph bulkUpdateGraph, ODataWriterWrapper odataWriter)
+        {
+            HashSet<Descriptor> relatedDescriptors = bulkUpdateGraph.GetRelatedDescriptors(entityDescriptor);
+            Dictionary<string, List<Descriptor>> groupedRelatedLinks = GroupRelatedLinksByNavigationProperty(relatedDescriptors);
+            ClientTypeAnnotation clientType = null;
+
+            if (clientType == null)
+            {
+                ClientEdmModel model = this.requestInfo.Model;
+                clientType = model.GetClientTypeAnnotation(model.GetOrCreateEdmType(entityDescriptor.Entity.GetType()));
+            }
+
+            foreach (KeyValuePair<string, List<Descriptor>> relatedLinks in groupedRelatedLinks)
+            {
+                LinkDescriptor relatedEnd = null;
+                ODataNestedResourceInfo navigationLink = new ODataNestedResourceInfo();
+                
+                //these items will all be of the same type ~ we'll be using the 

```

Comment:
Does this mean we can't have a mixture of `EntityDescriptor` and `LinkDescriptor` in `grouprdRelatedLinks`?

Reply:
we can but they will all be of the same `type`.